### PR TITLE
Add a test case for BuildInputOutputAliases

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This patch tries to add a test case for `BuildInputOutputAliases` and fixes the corresponding metric which reports a bogus value that's the total number of outputs.

Test Plan:
PJRT_DEVICE=CPU python test/test_input_output_aliases.py

### CODE REVIEW

**Comment by alanwaketan** [comment]
This is still a WIP as I'm puzzled on the way how we trigger `BuildInputOutputAliases` and then I still need to add it to run_tests.sh. @JackCaoG Can you provide more insights?

File: test/test_input_output_aliases.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import torch_xla.debug.metrics as met
+import unittest
+
+class MetricsTest(unittest.TestCase):
+
+  def test_non_view(self):
+    xla_device = xm.xla_device()
+    t1 = torch.randn(4, 2, 2).to(xla_device)
+    t2 = torch.randn(4, 2, 2).to(xla_device)
+    xm.mark_step()
```
Comment (alanwaketan) [inline test/test_input_output_aliases.py:15]: Here we trigger `BuildInputOutputAliases` and the counter is 2.

File: test/test_input_output_aliases.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    t3 = t1 + t2
+    t1 = t1 * 2
+    t2 = t2 + 2
+    xm.mark_step()
```
Comment (alanwaketan) [inline test/test_input_output_aliases.py:20]: Here we didn't.

**Comment by alanwaketan** [comment]
> @alanwaketan do you still plan to merge this pr?

Can we talk more about this tomorrow? I have some questions.

**Comment by JackCaoG** [comment]
Let me actually play with your code a bit, the way that `BuildInputOutputAliases` is written is confusing, I might need to add some comment to it or I will forget how it works very quickly.

**Comment by alanwaketan** [comment]
> Let me actually play with your code a bit, the way that `BuildInputOutputAliases` is written is confusing, I might need to add some comment to it or I will forget how it works very quickly.

Sounds good. Feel free to develop on the branch.

**Comment by JackCaoG** [comment]
Ah I know what's going on by
```
t2 = t2 + 2
```
you created an intermediate tensor with new tensor ID, if you do
```
t2 += 2
```
the alias count incremented as expected

**Comment by JackCaoG** [comment]
Let me rebase and add some comments to the code

**Comment by alanwaketan** [comment]
> Ah I know what's going on, by
> 
> ```
> t2 = t2 + 2
> ```
> 
> you created an intermediate tensor with new tensor ID and assign it back to t2, if you do
> 
> ```
> t2 += 2
> ```
> 
> the alias count incremented as expected

This is tricky. Thanks for the investigation!

**Review by JackCaoG — APPROVED** [review]
This is a bit weird, but I am going to approve my own change lol

**Comment by alanwaketan** [comment]
I guess we can land this now, and later I can add more test cases for views. Thanks for helping this, Jack.

**Comment by JackCaoG** [comment]
@alanwaketan yea I will merge this once all test passed.
